SemanticUI/Modules/Dropdown.elm

```diff
@@ -0,0 +1,434 @@
+            dropdown config
+    in
+    Dropdown { ordinaryConfig | variation = Button config.button }
+
+
+{-| Render a dropdown with the provided layout
+-}
+toCustomHtml : (Builder msg -> Html msg) -> Dropdown msg -> Html msg
+toCustomHtml layout (Dropdown config) =
+    case config.variation of
+        Ordinary ->
+            layout
+                { toToggle = toToggle config
+                , toDropdown = toRoot config
+                , drawer = drawer config
+                }
+
+        Button but ->
+            layout
+                { toToggle = toToggle config
+                , toDropdown =
+                    \element attrs ->
+                        Button.viewAs
+                            (toRoot config element
+                                |> HtmlBuilder.prependAttributes attrs
+                            )
+                            but
+                , drawer = drawer config
+                }
+
+
+{-| Render a dropdown with a simple default layout
+-}
+toHtml : { builder | items : List (Html msg) } -> Dropdown msg -> Html msg
+toHtml { items } dropdownControl =
+    let
+        layout builder =
+            builder.toDropdown div [] [ builder.drawer [] (List.map (toItem div [] << List.singleton) items) ]
+    in
+    toCustomHtml layout dropdownControl
+
+
+toRoot :
+    { config
+        | drawerState : Drawer.State
+        , identifier : String
+        , onToggle : Drawer.State -> msg
+        , attributes : List (Attribute msg)
+        , toggleEvent : Toggle.Event
+        , disabled : Bool
+        , dropdownIcon : Bool
+        , fluid : Bool
+        , scrolling : Bool
+    }
+    -> HtmlBuilder msg
+    -> HtmlBuilder msg
+toRoot config element attrs children =
+    let
+        isVisible =
+            Drawer.isVisible config.drawerState
+    in
+    Dropdown.root
+        { identifier = config.identifier
+        , onToggle = toDropdownOnToggle config.drawerState config.onToggle
+        , toggleEvent = toDropdownToggleEvent config.toggleEvent
+        , isToggled = Drawer.isToggled config.drawerState
+        }
+        element
+        (config.attributes
+            ++ (classList
+                    [ ( "ui", True )
+                    , ( "active", isVisible )
+                    , ( "visible", isVisible )
+                    , ( "disabled", config.disabled )
+                    , ( "fluid", config.fluid )
+                    , ( "scrolling", config.scrolling )
+                    , ( "dropdown", True )
+                    ]
+                    :: style "position" "relative"
+                    :: attrs
+               )
+        )
+        (toToggle config
+            div
+            [ style "position" "absolute"
+            , style "width" "100%"
+            , style "height" "100%"
+            , style "left" "0%"
+            , style "top" "0%"
+            ]
+            []
+            :: children
+            ++ (if config.dropdownIcon then
```

Comment:
It would be much better to allow specifying the icon. That's something we need, so a boolean is insufficient.

Reply:
@domenkozar Do you have a button dropdown? 

If so, the way this seems to work is that it's technically a "labeled icon button" that also happens to be a dropdown. So you have:

`ui ...(dropdown classes)... dropdown ....(button classes)... button`

so for the icon:

`ui ...(dropdown classes)... dropdown labeled icon button`

From what I can see, SemanticUI appears to treat the `dropdown icon` in a special way - you can't seem to replace it directly without encountering layout bugs.

Here's how I'm doing it for a filter button at the moment:

```elm
            Selection.button
                { button =
                    Button.init
                        |> Button.icon (Just Icon.Filter)
                , identifier = "filter-by"
                , onToggle = FilterByDrawerMsg
                , onSelect = SetFilterBy
                , drawerState = model.filterByDrawerState
                , currentSelection = Just model.selectedFilter
                , defaultLabel = text "Select filter"
                , selectionLabel = text << filterToString
                }
                |> Selection.toHtml
                    { options = allFilters
                    , optionLabel = text << filterToString
                    }
```

Does that support your use-case?

Reply:
I guess I'll play around once you have this ready and report back. Need to try to really say it works.